packages/fuchsia_remote_debug_protocol/lib/src/runners/ssh_command_runner.dart

```diff
@@ -0,0 +1,99 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' show ProcessResult;
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../common/logging.dart';
+import '../common/network.dart';
+
+/// An error raised when a command fails to run within the [SshCommandRunner].
+///
+/// Note that this occurs for both connection failures, and for failure to
+/// running the command on the remote device. This error is raised when the
+/// subprocess running the SSH command returns a nonzero exit code.
+class SshCommandError extends Error {
+  /// The reason for the command failure.
+  final String message;
+
+  /// Basic constructor outlining the reason for the SSH command failure through
+  /// the message string.
+  SshCommandError(this.message);
+
+  @override
+  String toString() {
+    return 'SshCommandError: $message\n${super.stackTrace}';
```

Comment:
hint: You can $ in front of SshCommandError, which will produce the same string but protect from spelling mistakes or future class renames.